listenbrainz/webserver/static/js/src/playlists/utils.tsx

```diff
@@ -65,6 +65,27 @@ export function millisecondsToStr(milliseconds: number) {
   return "< 1s";
 }
 
+// Originally by Sinjai https://stackoverflow.com/a/67462589
+export function msToTime(milliseconds: number) {
```

Comment:
I understand your point. I should have explained why I introduced a new method over the existing one. `millisecondsToStr` was what I wanted to use originally.
- Yes, my concern was padding of the numbers, but also that seconds have `s` suffix in [the method](https://github.com/metabrainz/listenbrainz-server/blob/master/listenbrainz/webserver/static/js/src/playlists/utils.tsx#L54).
- And even if we were to get rid of that trailing `s`, another problem was that it shows `60s` after `59s` instead of `1:00` and it shows `<1s` when the song starts playing to indicate 0 seconds.

I think `millisecondsToStr` was more intended to be used for the `ListenCard` timestamp.

Therefore, instead of making any changes in the existing method, I decided to introduce a new one. I should have probably discussed this with you earlier.

Do you want me to modify the `millisecondsToStr` method to make it work both for timestamp as well as elapsed time purposes?

Reply:
Thanks for the explanation, that makes more sense now.
The only uses I can see in the codebase for the existing `millisecondsToStr` are for track duration shown in the ListenCard and some other components.

Consequently, I think we should modify the existing method to return something closer to what you have done with the new `msToTime`, because 1. we should harmonize across the codebase and 2. you're right, for track durations this isn't standard:
<img width="723" alt="image" src="https://user-images.githubusercontent.com/6179856/169545328-c806d2df-e747-4e5b-b08d-c7cfe332f439.png">

There is the possibility that we would have durations spanning days (for example playlist total durations), but I think it just makes more sense to stop at hours.

So for an extended example, you'd have:
00:01 - one second
01:02:00 - one hour and 2 minutes
28:00:00 - 2+ days


Sooooo… really that's the method which you have implemented if I'm not mistaken? :D
In which case, please go ahead and replace the old method with your new method, but keeping the old function name so that it'll be used across the codebase.

Reply:
With the proper credit comment at the top :)
If you end up using a bit of both functions, please put both attributions.